hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoSqlTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.sql.impl.connector.mongodb;
+
+import com.hazelcast.internal.util.EmptyStatement;
+import com.hazelcast.jet.sql.SqlTestSupport;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlService;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.testcontainers.containers.MongoDBContainer;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class})
+public abstract class MongoSqlTest extends SqlTestSupport {
+    private static final String TEST_MONGO_VERSION = System.getProperty("test.mongo.version", "6.0.3");
+
+    @ClassRule
+    public static final MongoDBContainer mongoContainer
+            = new MongoDBContainer("mongo:" + TEST_MONGO_VERSION);
+
+    protected static SqlService sqlService;
+    protected static MongoClient mongoClient;
+    protected static MongoDatabase database;
+    protected static String databaseName;
+
+    @Rule
+    public final TestName testName = new TestName();
+
+    @BeforeClass
+    public static void beforeClass() {
+        initialize(1, null);
+        sqlService = instance().getSql();
+        mongoClient = MongoClients.create(mongoContainer.getConnectionString());
+        databaseName = randomName();
+        database = mongoClient.getDatabase(databaseName);
+    }
+
+    @AfterClass
+    public static void close() {
+        if (mongoClient != null) {
+            mongoClient.close();
+        }
+    }
+
+    protected void execute(String sql, Object... arguments) {
+        try (SqlResult ignored = sqlService.execute(sql, arguments)) {
+            EmptyStatement.ignore(null);
+        }
+    }
+
+    protected static String options() {
+        return String.format("OPTIONS ( 'database' = '%s', 'connectionString' = '%s') ", databaseName,
+                mongoContainer.getConnectionString());
+    }
+
+    protected String methodName() {
+        return testName.getMethodName();
+    }
```

Comment:
Can you explain this?

Reply:
For test separation I'm inserting always to a collection with name the same as test method name - so it's guaranteed, that one test will never depend on other test

Reply:
We typically use `com.hazelcast.jet.sql.SqlTestSupport#randomName` for that.

Reply:
95d588f0f37cf4ff88edff5f638e4f2bcc9fa7f6